tests/p2p_connection_test.rs

```diff
@@ -6,46 +6,124 @@ mod setup;
+    use std::time::Duration;
 
     use crate::setup::NodeHandle;
     use uniffi_lipalightninglib::config::NodeAddress;
 
+    const NIGIRI_LND_ADDR: &str = "127.0.0.1:9735";
+    const FAULTY_ADDR: &str = "127.0.0.1:9";
+
     #[test]
     fn test_successful_p2p_connection() {
         setup::nigiri::start();
         let lsp_info = setup::nigiri::query_lnd_info().unwrap();
         let lsp_node = NodeAddress {
             pub_key: lsp_info.pub_key,
-            address: "127.0.0.1:9735".to_string(),
+            address: NIGIRI_LND_ADDR.to_string(),
         };
 
-        let node = NodeHandle::new(lsp_node).start().unwrap();
-        assert_eq!(node.get_node_info().num_peers, 1)
+        let node = NodeHandle::new(lsp_node.clone()).start().unwrap();
+
+        assert_eq!(node.get_node_info().num_peers, 1);
+        assert!(node.connected_to_node(&lsp_node));
     }
 
     #[test]
     fn test_failing_p2p_connection() {
+        setup::nigiri::start();
         let lsp_info = setup::nigiri::query_lnd_info().unwrap();
         let lsp_node = NodeAddress {
             pub_key: lsp_info.pub_key,
-            address: "127.0.0.1:9".to_string(),
+            address: FAULTY_ADDR.to_string(),
         };
-        assert!(NodeHandle::new(lsp_node).start().is_err());
+
+        let node = NodeHandle::new(lsp_node.clone()).start().unwrap();
+
+        assert_eq!(node.get_node_info().num_peers, 0);
+        assert!(!node.connected_to_node(&lsp_node));
+    }
+
+    #[test]
+    fn test_start_node_while_lsp_is_down() {
+        let lsp_info = setup::nigiri::query_lnd_info().unwrap();
+        let lsp_node = NodeAddress {
+            pub_key: lsp_info.pub_key,
+            address: NIGIRI_LND_ADDR.to_string(),
+        };
+
+        // Let's shutdown LND, while we leave Esplora running (test should not affect chain sync)
+        setup::nigiri::pause();
+        setup::nigiri::resume_without_ln();
+
+        let node = NodeHandle::new(lsp_node.clone()).start().unwrap();
+
+        // Wait for the LDK to connect to the LSP (which should fail)
+        sleep(Duration::from_millis(1500));
+
+        assert_eq!(node.get_node_info().num_peers, 0);
+        assert!(!node.connected_to_node(&lsp_node));
+
+        // Now let's start Nigiri with LND again
+        setup::nigiri::pause();
+        setup::nigiri::resume();
+
+        // Wait for the LDK to connect to the LSP (which should now succeed)
+        sleep(Duration::from_millis(1500));
+
+        assert_eq!(node.get_node_info().num_peers, 1);
+        assert!(node.connected_to_node(&lsp_node));
+    }
+
+    #[test]
+    fn test_stop_node_while_lsp_is_down() {
+        setup::nigiri::start();
+        let lsp_info = setup::nigiri::query_lnd_info().unwrap();
+        let lsp_node = NodeAddress {
+            pub_key: lsp_info.pub_key,
+            address: NIGIRI_LND_ADDR.to_string(),
+        };
+
+        {
+            let node = NodeHandle::new(lsp_node.clone()).start().unwrap();
+
```

Comment:
In case we remove the sleep from `start()`, we need to add it here.

Reply:
Obsolete (see discussion above)